### PR TITLE
Improve performance when using checkout-ui-custom

### PR DESCRIPTION
**Pull request by wender**
### Changed
- When using `checkout-ui-custom`, it will first look for files on **VBase**, **Masterdata** is now a fallback

### CODE REVIEW

File: node/middlewares/settings.ts [hunk 1]
```diff
@@ -40,32 +42,42 @@ export async function getSettingsFromContext(ctx: Context, next: () => Promise<a
...
-          },
-        })
-
-        if (mdFiles && mdFiles.length) {
-          settingFile = mdFiles[0][field]
-        } else {
-          settingFile = allSettingsFromDeclarer[file]
+      const vbFile = await vbase.getFile('checkoutuicustom', `${workspace}-${field}`)
+        .then((res: any) => {return res.data})
+        .catch(() => null)
```
Comment (brunoabreu) [inline node/middlewares/settings.ts:49]: We should at least log something on Splunk.

File: node/middlewares/settings.ts [hunk 2]
```diff
@@ -40,32 +42,42 @@ export async function getSettingsFromContext(ctx: Context, next: () => Promise<a
...
-            page: 1,
-            pageSize: 1,
-          },
-        })
-
-        if (mdFiles && mdFiles.length) {
-          settingFile = mdFiles[0][field]
-        } else {
-          settingFile = allSettingsFromDeclarer[file]
+      const vbFile = await vbase.getFile('checkoutuicustom', `${workspace}-${field}`)
```
Comment (brunoabreu) [inline node/middlewares/settings.ts:47]: Is it a JSON? Could you use getJSON?

**Review by brunoabreu — CHANGES_REQUESTED** [review]

File: node/middlewares/settings.ts (hunk 2 above)
Comment (wender) [reply node/middlewares/settings.ts:47]: It's not a JSON, it's either a CSS or a Javascript file

File: node/middlewares/savevb.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  const { workspace, field, file } = body
+
+  const stream = new Readable()
+  stream._read = () => undefined
+  stream.push(file)
+  stream.push(null)
+
+  await vbase
+    .saveFile('checkoutuicustom', `${workspace}-${field}`, stream)
```
[the remark this answers is not in the document]
Comment (wender) [reply node/middlewares/savevb.ts:20]: it's either a CSS or a Javascript file

**Review by brunoabreu — APPROVED** [review]

**Pull request merged**